src/intervals/interval_operations/boolean.jl

```diff
@@ -0,0 +1,228 @@
+    return false
+end
+
+"""
+    ⊆(a,b)
+
+Checks if all the points of the interval `a` are within the interval `b`.
+
+Typed with \\subseteq<TAB>.
+
+Implement the `subset` function of the IEEE Std 1788-2015 (Table 9.3).
+"""
+function ⊆(a::Interval, b::Interval)
+    isempty(a) && return true
+    b.lo ≤ a.lo && a.hi ≤ b.hi
+end
+
+"""
+    ⊂(a,b)
+
+Checks if `a` is a strict subset of interval `b`.
+
+Typed with \\subset<TAB>.
+"""
+function ⊂(a::Interval, b::Interval)
+    a ≛ b && return false
+    return a ⊆ b
+end
+
+⊇(a::Interval, b::Interval) = b ⊆ a
+⊃(a::Interval, b::Interval) = b ⊂ a
+
+"""
+    isweaklyless(a, b)
+
+Checks if the interval `a` is weakly less than interval `b`.
+
+Note that this is not equivalent as saying every element of `a` is less than
+any element of `b`.
+
+Implement the `less` function of the IEEE Std 1788-2015 (Table 10.3).
+"""
+function isweaklyless(a::F, b::F) where {F<:Interval}
+    isempty(a) && isempty(b) && return true
+    (isempty(a) || isempty(b)) && return false
+    (a.lo ≤ b.lo) && (a.hi ≤ b.hi)
+end
+
+"""
+    precedes(a, b)
+
+Checks if the interval `a` is to the left of interval `b`.
+
+Implement the `precedes` function of the IEEE Std 1788-2015 (Table 10.3).
+"""
+function precedes(a::F, b::F) where {F<:Interval}
+    (isempty(a) || isempty(b)) && return true
+    return a.hi ≤ b.lo
+end
+
+"""
+    isinterior(a,b)
+
+Checks if all the points of the interval `a` are within the interior of
+interval `b`.
+
+Implement the `interior` function of the IEEE Std 1788-2015 (Table 9.3).
+"""
+function isinterior(a::Interval, b::Interval)
+    isempty(a) && return true
+    return isweaklylessprime(b.lo, a.lo) && isweaklylessprime(a.hi, b.hi)
+end
+
+"""
+    isstrictless(a, b)
+
+Checks if the interval `a` is strictly less than interval `b`, which is true
+if `a.lo < b.lo` and `a.hi < b.hi`.
+
+For variants in the definition of "strictly less than" for intervals see
+`strictprecedes` and `<`.
+
+Implement the `strictLess` function of the IEEE Std 1788-2015 (Table 10.3).
+"""
+function isstrictless(a::F, b::F) where {F<:Interval}
+    isempty(a) && isempty(b) && return true
+    (isempty(a) || isempty(b)) && return false
+    return isweaklylessprime(a.lo, b.lo) && isweaklylessprime(a.hi, b.hi)
+end
+
+"""
+    strictprecedes(a, b)
+
+Checks if the interval `a` is strictly to the left of interval `b`.
+
+Implement the `strictPrecedes` function of the IEEE Std 1788-2015 (Table 10.3).
+"""
+function strictprecedes(a::F, b::F) where {F<:Interval}
+    (isempty(a) || isempty(b)) && return true
+    return a.hi < b.lo
+end
+
+"""
+    isdisjoint(a,b)
+
+Checks if all the points of the interval `a` are within the interior of
+interval `b`.
+
+Implement the `disjoint` function of the IEEE Std 1788-2015 (Table 9.3).
+"""
+function isdisjoint(a::Interval, b::Interval)
+    (isempty(a) || isempty(b)) && return true
+    return isweaklylessprime(b.hi, a.lo) || isweaklylessprime(a.hi, b.lo)
```

Comment:
I think this line is equivalent to `isempty(a ∩ b)`, but I'm not sure which one is faster